synapse/cortex.py

```diff
@@ -6502,6 +6510,9 @@ async def addCronJob(self, cdef):
             The incunit if not None it must be larger in unit size than all the keys in all reqs elements.
             Non-recurring jobs may also have a req of 'now' which will cause the job to also execute immediately.
         '''
+        if (user := cdef.pop('user', None)) is not None:
+            cdef['creator'] = user
```

Comment:
Just a note to clarify here, we make sure the older `creator` prop is populated in the top half of the handler and switch it to `user` in the bottom half of the nexus handler to ensure compatibility when mirrors are updated before the leader.

Reply:
Does this mean that the nexus entries will always contain the creator key?

Reply:
Yeah the nexus entries will have the creator key until we remove this shim

Reply:
The other option we could do would be to increment the cell nexus version and only do this if we aren't on the newer nexus version.

Reply:
Lets do that so we have a path to eventually phasing it out :)